urlb44e9843ba_QuasimodoNZ_SWEN222_Game_Project_tgz-pJ8-Game_World_GameWorldLevelBuilderJ8/src/Game_World/Avatar.java

```diff
@@ -0,0 +1,406 @@
+package Game_World;
+
+import java.awt.Graphics;
+import java.awt.Image;
+import java.awt.Point;
+import java.awt.Rectangle;
+import java.util.ArrayList;
+import java.util.List;
+
+import Game_World_Objects.Food;
+import Game_World_Objects.Torch;
+import Game_World_Objects.Trap;
+import Graphics_Renderer.GraphicsRenderer;
+import Graphics_Renderer.Tileset;
+import Object_Interfaces.Containable;
+import Object_Interfaces.Container;
+import Object_Interfaces.GameObject;
+
+public class Avatar implements Container, GameObject {
+	/**
+	 * The Class represents the Avatar object that represents the Player in the GameWorld.
+	 * The Avatar is a Container (Avatar has a backpack) which can hold up to 16 items.
+	 * The Avatar can move and interact with the GameWorld.
+	 *
+	 * @author Danesh Abeyratne
+	 *
+	 */
+	private String name;// the name of the player
+	private String playerSprite;
+	private Player owner;// the player object for this Avatar
+	private List<Containable> posessions;
+	private int playerID;
+	private Rectangle boundingBox;
+	private GameWorld world;
+	private int stepState = 1; // 0 = Idle and 1 through 3 = mid step
+	private int direction = 0; // 0 = South, 1 = West, 2 = North and 3 = East
+	private int health = 100; // the avatar's initial health state
+	private Location currentLocation;
+	private String locationName;
+	private int stepsize = 4;// the number of pixels the avatar moves
+
+	public Avatar(String name, int numPlayers) {
+		this.name = name;
+		int num = numPlayers % 4 + 1;
+		playerSprite = "player" + num; // randomNum;
+		boundingBox = new Rectangle(200 + (numPlayers * 100), 200, 37, 9);
+		posessions = new ArrayList<Containable>(16);
+
+	}
+
+	public Avatar() {} // default constructor for XML
+
+	/**
+	 * Move command dependent on keystroke
+	 *
+	 * @param int comd
+	 */
+	public void move(int comd) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 63-78 into a separate method.